grakn-test/src/test/java/ai/grakn/test/engine/tasks/connection/RedisConnectionTest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016  Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.test.engine.tasks.connection;
+
+import ai.grakn.concept.ConceptId;
+import ai.grakn.engine.tasks.connection.RedisConnection;
+import ai.grakn.test.EngineContext;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * <p>
+ *    Tests that the connection to redis is working and that data can be persisted and retrieved into/from redis.
+ * </p>
+ *
+ * @author fppt
+ */
+public class RedisConnectionTest {
+
+    @ClassRule
+    public static final EngineContext engine = EngineContext.startInMemoryServer();
```

Comment:
Yeah, when engine starts we definitely need reds running. I actually meant that if we have a separate Redis context, you wouldn't need to start the entirety of engine for this test that does not need engine.